### PR TITLE
chore: remove punycode warning docs

### LINKED ISSUE #811: node v22 has a punycode depre
### Describe what should be investigated or refactored

On node v22 there is a deprecation warning for `punycode`.

```bash
(node:47514) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
```

**Definition of done:**
Remove the punycode warning

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

The punycode deprecation warning is intrusive and very ugly in the cli. We need to remove it yet we do not want to suppress all warnings incase there are relevant warnings. This adds documentation around how to suppress that exact punycode warning 

## Related Issue

Fixes #

Relates to #811 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

**Comment by cmwylie19** [comment]
check the blast radius and make sure it does not suppress any other warnings other than punycode

**Comment by cmwylie19** [comment]
> check the blast radius and make sure it does not suppress any other warnings other than punycode

seems to be fine

image

**Comment by cmwylie19** [comment]
we can document to set `export NODE_OPTIONS="--disable-warning=DEP0040"` in the docs under FAQ how do i remove punycode warning and

File: src/cli.ts [hunk 1]
```diff
@@ -19,7 +19,10 @@ import kfc from "./cli/kfc";
 if (process.env.npm_lifecycle_event !== "npx") {
   console.warn("Pepr should be run via `npx pepr <command>` instead of `pepr <command>`.");
 }
-
+// Remove punycode warning
+if (process.env.PEPR_MODE !== "dev") {
+  process.removeAllListeners("warning");
+}
```
Comment (cmwylie19) [inline src/cli.ts]:
```suggestion
```

Comment (cmwylie19) [reply src/cli.ts]: this does nothing bc it is invoked too late for it to be effective

File: src/cli.ts [hunk 2]
```diff
@@ -19,7 +19,6 @@ import kfc from "./cli/kfc";
 if (process.env.npm_lifecycle_event !== "npx") {
   console.warn("Pepr should be run via `npx pepr <command>` instead of `pepr <command>`.");
 }
-
 const program = new RootCmd();
```
Comment (schaeferka) [inline src/cli.ts]:
```suggestion

const program = new RootCmd();
```

**Review by schaeferka — COMMENTED** [review]
See minor comment

**Review by schaeferka — APPROVED** [review]

**Pull request merged**